src/engraving/dom/gradualtempochange.cpp

```diff
@@ -299,6 +309,75 @@ Sid GradualTempoChangeSegment::getPropertyStyle(Pid id) const
     return TextLineBaseSegment::getPropertyStyle(id);
 }
 
+GradualTempoChangeSegment* GradualTempoChangeSegment::findElementToSnapBefore() const
+{
+    const System* sys = system();
+    IF_ASSERT_FAILED(sys) {
+        return nullptr;
+    }
+
+    GradualTempoChange* gradTempoChange = tempoChange();
+    Fraction startTick = gradTempoChange->tick();
+    if (!sys->measures().empty() && startTick == sys->measures().front()->tick()) {
+        return nullptr;
+    }
+
+    auto intervals = score()->spannerMap().findOverlapping(startTick.ticks(), startTick.ticks());
+    for (auto interval : intervals) {
+        Spanner* spanner = interval.value;
+        if (!spanner->isGradualTempoChange() || spanner->track() != gradTempoChange->track() || spanner->tick2() != startTick
```

Comment:
I'd recommend avoiding such complicated conditions (I've seen a few of them in this PR). They're quite difficult to read/debug

Reply:
Done my best to break them down into more readable pieces 👍